src/traces/heatmap/hover.js

```diff
@@ -96,8 +96,8 @@ module.exports = function hoverPoints(pointData, xval, yval, hovermode, contour)
     if(zmask && !zmask[ny][nx]) zVal = undefined;
 
     var text;
-    if(Array.isArray(trace.text) && Array.isArray(trace.text[ny])) {
-        text = trace.text[ny][nx];
+    if(Array.isArray(cd0.text) && Array.isArray(cd0.text[ny])) {
+        text = cd0.text[ny][nx];
```

Comment:
> and only keep the re-dimensioned text array in calcdata.

I agree 100%. But, I wanted to keep this patch to a minimum as @rreusser's carpet PR already alters [`heatmap/convert_column_xyz.js`](https://github.com/plotly/plotly.js/pull/1239/files#diff-9f910383868c78bde87fcb8d3905fe91R14).

I'll make a comment there.